web/data/data.go

```diff
@@ -162,7 +173,7 @@ func UpdateDoc(c echo.Context) error {
 
 	doc.Type = c.Param("doctype")
 
-	if err := CheckWritable(doc.Type); err != nil {
+	if err := hooks.Run(instance, realtime.EventUpdate, &doc, nil); err != nil {
```

Comment:
And why this call happen before checking that the doc is valid and the permissions are OK?